crop-planner/src/Table.tsx

```diff
@@ -45,27 +46,50 @@ export default class Table extends PureComponent<Props, State> {
       {sortBy(this.state.fields, field => field.name).map(field => this.renderFieldRow(field))}
     </div>
 
+  humusBalanceIcon = (balance: number) => {
+    return balance && balance > 0 ? 'table__cell--positive' : 'table__cell--negative'
+  }
+
   renderFieldRow = (field: Field) =>
-    <div className="table__row" key={field.id}>
+    <div className="table__row" key={field.id.toString()}>
       <div className="table__cell">{field.name}</div>
       <div className="table__cell table__cell--right">{field.area}</div>
 
       {sortBy(field.crops, crop => crop.year).map(seasonalCrop => this.renderCropCell(field, seasonalCrop))}
 
-      <div className="table__cell table__cell--right">--</div>
+      <div className={"table__cell table__cell--right " + this.humusBalanceIcon(field.humus_balance)}>
+         { field.humus_balance || '--' }
+      </div>
     </div>
 
   renderCropCell = (field: Field, seasonalCrop: SeasonalCrop) =>
-    <div className="table__cell table__cell--center table__cell--with-select">
+    <div className="table__cell table__cell--center table__cell--with-select" key={seasonalCrop.year.toString()}>
       <CropSelect
         selectedCrop={seasonalCrop.crop}
         allCrops={this.state.allCrops}
         onChange={newCrop => this.changeFieldCrop(newCrop, field.id, seasonalCrop.year)}
       />
     </div>
 
-  changeFieldCrop = (newCrop: Crop | null, fieldId: number, cropYear: number) =>
+  calculateHumusBalance = async (fieldId: number) => {
+    const currentField = this.state.fields.find(field => field.id === fieldId)
+
+    if (currentField) {
+      const humus_deltas = sortBy(currentField.crops, crop => crop.year)
+        .map(seasonCrop => seasonCrop.crop ? seasonCrop.crop.humus_delta : 0)
+      const { humus_balance } = await fetchHumusBalance(humus_deltas)
```

Comment:
Instead of sendinf humus deltas to backend it would be preferable to send field new crop ids. Backend can then do the calculations and send updated humus balances back.